contracts/modules/token/TokenController.sol

```diff
@@ -508,6 +608,50 @@ contract TokenController is IERC1132, Iupgradable {
     emit Unlocked(_of, _reason, _amount);
   }
 
+  function withdrawCoverNote(
+    address _of,
+    bytes32[] calldata _reasons,
+    uint[] calldata _indexes
+  ) external onlyInternal {
+
+    uint reasonCount = lockReason[_of].length;
+    require(reasonCount > 0, "TokenController: no locked cover notes found");
+
+    uint lastReasonIndex = reasonCount.sub(1);
+    uint totalAmount = 0;
+
+    // The iteration is done from the last to first to prevent reason indexes from
+    // changing due to the way we delete the items (copy last to current and pop last).
+    // The provided indexes array must be ordered, otherwise reason index checks will fail.
+
+    for (uint i = _reasons.length; i > 0; i--) {
+
+      bytes32 _reason = _reasons[i - 1];
+      uint _resonIndex = _indexes[i - 1];
+      require(lockReason[_of][_resonIndex] == _reason, "TokenController: bad reason index");
+
+      uint amount = locked[_of][_reason].amount;
+      require(amount != 0, "TokenController: amount is zero");
+
+      totalAmount = totalAmount.add(amount);
+      locked[_of][_reason].amount = 0;
+
+      if (lastReasonIndex != _resonIndex) {
+        lockReason[_of][_resonIndex] = lockReason[_of][lastReasonIndex];
+      }
+
+      lockReason[_of].pop();
+      emit Unlocked(_of, _reason, amount);
+
+      if (i > 1) {
+        // if i > 1 then the input has more entries but there are none left in the lockReason array
```

Comment:
would make this more clear

"reduce the array size (remove last element). if no items left in _lockReasons, _reasons contains too many items" 

something like that